xchange-coinbasepro/src/main/java/org/knowm/xchange/coinbasepro/CoinbaseProExchange.java

```diff
@@ -110,6 +111,11 @@ public ExchangeSpecification getDefaultExchangeSpecification() {
     throw new UnsupportedOperationException("CoinbasePro uses timestamp rather than a nonce");
   }
 
+  @Override
+  public ResilienceRegistries getResilienceRegistries() {
+    return CoinbaseProResilience.INSTANCE;
+  }
```

Comment:
it would be better to do lazy initialization as suggested in: https://github.com/knowm/XChange/wiki/Implementing-resiliency-features#step-1

Reply:
My only issue with the lazy initialization is that people can create multiple instances of the resilience class, is that a property we want, I figured for the rate limiters we want a singleton?

Reply:
doing lazy initialization in the way suggested in https://github.com/knowm/XChange/wiki/Implementing-resiliency-features#step-1

```
public class BinanceExchange extends BaseExchange {

  private static ResilienceRegistries RESILIENCE_REGISTRIES;

  // ...

  @Override
  public ResilienceRegistries getResilienceRegistries() {
    if (RESILIENCE_REGISTRIES == null) {
      RESILIENCE_REGISTRIES = BinanceResilience.createRegistries();
    }
    return RESILIENCE_REGISTRIES;
  }
```

will result in a singelton - ` BinanceResilience.createRegistries` will be called only once, when the first exchange instance gets created - any more instances of the exchange will result in using the previously created ResilienceRegistries instance

Reply:
It will be a singleton from the BinanceExchange perspective, but it will not be a singleton from the JVM perspective. But this is less likely so I can change it if you prefer.

Reply:
ResilienceRegistries will not be a singelton from the JVM perspective in either case 

CoinbaseProResilience is also not a singleton, its an utility class

because of this `CoinbaseProResilience.INSTANCE` just looks confusing to me

we want all instances of `XExchange` to use a shared `ResilienceRegistries` - if think that this should be expressed in the implementation of `XExchange`